### PR TITLE
update category query

### PR DESCRIPTION
**Pull request by matthewdking**

### CODE REVIEW

**Comment by matthewdking** [comment]
please review and leave comments. The query needs to do something with the response I just wasn't sure what :confused:

**Comment by Karyum** [comment]
@matthewdking you could always pass the response from the controller to the query and use it to render a page :wink:

**Review by Wyrd00 — COMMENTED** [review]
sweet as

**Review by sajedazoabi — APPROVED** [review]

**Comment by sajedazoabi** [comment]
Good Job @matthewdking

**Review by Wyrd00 — APPROVED** [review]

**Comment by matthewdking** [comment]
@Karyum I think that's what we are going to try to do! Can you merge it so we can have all the queries and then we can work from there?

File: src/queries/queries.js [hunk 1]
```diff
@@ -8,7 +8,59 @@ const getQuestion = (category) => {
...
 };
 
+const updateCategoryScore = (category, mentorId1, mentordId2, mentorScore1, mentorScore2,) = {
+
+  const updateRating = (category, mentorScore, mentorId, callback) => {
+    db.query(`UPDATE $1 SET rating = rating + $2 WHERE mentor_id = $3`, [category, mentorScore, mentorId], (error, result) => {
+      if (error) {
+        callback(error);
+      } else {
+        return callback(result);
```
Comment (Karyum) [inline src/queries/queries.js]: isn't it error first callback ?

File: src/queries/queries.js [hunk 2]
```diff
@@ -8,7 +8,59 @@ const getQuestion = (category) => {
...
+      }
+    });
+  }
+
+  const updateRespectiveScore = (category, mentorName, mentorScore, mentorId, callback) => {
+    db.query(`UPDATE $1 SET $2 = $2 + $3 WHERE mentor_id = $4 RETURNING $2`, [category, mentorName, mentorScore, mentorId], (error, result) => {
+      if (error) {
+        callback(error);
+      } else {
+        return callback(result);
```
Comment (Karyum) [inline src/queries/queries.js]: here too

File: src/queries/queries.js [hunk 3]
```diff
@@ -8,7 +8,59 @@ const getQuestion = (category) => {
...
+      }
+    });
+  }
+
+  updateRating(category, mentorScore1, mentorId1, (error, result) => {
+    if (error) {
+      console.log(error);
```
Comment (Karyum) [inline src/queries/queries.js]: so lets say there is an error whats going to happen?
so also here for example you could use the response from the controller to render the error page

File: src/queries/queries.js [hunk 4]
```diff
@@ -8,7 +8,59 @@ const getQuestion = (category) => {
...
+    } else {
+      updateRating(category, mentorScore2, mentorId2, (error, result) => {
+        if (error) {
+          console.log(error);
```
Comment (Karyum) [inline src/queries/queries.js]: also here

File: src/queries/queries.js [hunk 5]
```diff
@@ -8,7 +8,59 @@ const getQuestion = (category) => {
...
+        } else {
+          updateRespectiveScore(category, mentorName2, mentorScore1, mentorId1, (error, result) => {
+            if (error) {
+              console.log(error);
```
Comment (Karyum) [inline src/queries/queries.js]: here

File: src/queries/queries.js [hunk 6]
```diff
@@ -8,7 +8,59 @@ const getQuestion = (category) => {
...
+            } else {
+              //DO SOMETHING WITH RESULT
+              updateRespectiveScore(category, mentorName1, mentorScore2, mentorId2, (error, result) => {
+                if (error) {
+                  console.log(error);
```
Comment (Karyum) [inline src/queries/queries.js]: here

**Review by Karyum — CHANGES_REQUESTED** [review]

File: src/queries/queries.js (hunk 1 above)
Comment (matthewdking) [reply src/queries/queries.js]: good spot :eyes:

File: src/queries/queries.js (hunk 3 above)
Comment (matthewdking) [reply src/queries/queries.js]: I think we will add this later because we are still a bit unsure about what is happening where

**Comment by Karyum** [comment]
@matthewdking are you going to add the response to this PR or is it going to be a different one ??
cause it makes sense to be this PR

**Comment by matthewdking** [comment]
@Karyum can you check that out?

**Comment by Karyum** [comment]
@matthewdking so whats going to happen next with that object??

**Comment by matthewdking** [comment]
@Karyum its going to be used somewhere in a controller or view I think!

**Comment by matthewdking** [comment]
:pray:

**Review by Karyum — APPROVED** [review]

**Comment by Karyum** [comment]
@matthewdking you could use the controller's response but this works

**Pull request merged**